packages/mui-material/src/Accordion/Accordion.js

```diff
@@ -165,6 +168,20 @@ const Accordion = React.forwardRef(function Accordion(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
+  const backwardCompatibleSlots = { transition: TransitionComponentProp, ...slots };
+  const backwardCompatibleSlotProps = { transition: TransitionPropsProp, ...slotProps };
+
+  const [TransitionSlot, transitionProps] = useSlot('transition', {
+    elementType: Collapse,
+    externalForwardedProps: {
+      slots: backwardCompatibleSlots,
+      slotProps: backwardCompatibleSlotProps,
+    },
+    ownerState,
+  });
+
+  delete transitionProps.ownerState;
```

Comment:
The Transition component doesn't handle `transitionProps.ownerState` well, and it has no effect anyways.

Reply:
I am curious to see long term if we would have issues about this (it doesn't change the behavior so I am not worried about it, but if people use custom transitions they may expect to be able to have access to the `ownerState`. This is likely an edge case tough.

Reply:
I figured out why this was breaking. If we provide `transitionProps.ownerState`, then:

- The transition component, Collapse in this case, receives an `ownerState` prop
- [The `ownerState` property goes into `...other`](https://github.com/mui/material-ui/blob/b1ea9705596d1a208cb8a426172da0e250a4806d/packages/mui-material/src/Collapse/Collapse.js#L121)
- [The `other` is provided to Transition](https://github.com/mui/material-ui/blob/b1ea9705596d1a208cb8a426172da0e250a4806d/packages/mui-material/src/Collapse/Collapse.js#L278)
- [Transition does not recognize the `ownerState` prop, so it forwards it as `childProps`](https://github.com/reactjs/react-transition-group/blob/master/src/Transition.js#L385)
- [`childProps.ownerState` overrides Collapse's own `ownerState`, and the animation breaks.
](https://github.com/mui/material-ui/blob/b1ea9705596d1a208cb8a426172da0e250a4806d/packages/mui-material/src/Collapse/Collapse.js#L297)

This only happens on the initial open state as for subsequent enter calls, [the size in handled on the `onEntered` event](https://github.com/mui/material-ui/blob/b1ea9705596d1a208cb8a426172da0e250a4806d/packages/mui-material/src/Collapse/Collapse.js#L211).

That's why [the Tooltip cannot use Collapse as an animation](https://codesandbox.io/p/sandbox/tooltip-with-collapse-j269d2?file=%2Fsrc%2FDemo.tsx%3A13%2C5), as it provides [`ownerState` to the transition component](https://github.com/mui/material-ui/blob/b1ea9705596d1a208cb8a426172da0e250a4806d/packages/mui-material/src/Tooltip/Tooltip.js#L653). The Collapse transition has this issue and other transitions don't as the others don't rely on an inner `ownerState`, so accidentally overriding it doesn't have any effect.

I would recommend going with `delete transitionProps.ownerState` for now, opening a separate issue, and linking that issue to the code here so we remember to update it when we fix it.

What do you think?

Reply:
> I would recommend going with delete transitionProps.ownerState for now, opening a separate issue, and linking that issue to the code here so we remember to update it when we fix it.
> 
> What do you think?

Sounds like a good plan. Maybe the transition components should know to ignore the ownerstate - but that is a different discussion.

Reply:
Thanks for the detailed explanation, it was really helpful to understand at a glance @DiegoAndai